src/components/Button/Button.tsx

```diff
@@ -0,0 +1,168 @@
+import { ReactElement, ReactNode, useMemo } from 'react'
+import { Button as AntButton } from 'antd'
+import classnames from 'classnames'
+
+import { ButtonHierarchies, ButtonIconPositions, ButtonShapes, ButtonSizes, ButtonTypes } from './Button.types'
+import styles from './Button.module.css'
+
+const { Primary, Neutral, Danger } = ButtonHierarchies
+const { Left, Right } = ButtonIconPositions
+const { Square } = ButtonShapes
+const { Small, Middle, Large } = ButtonSizes
+const { Filled, Ghost } = ButtonTypes
+
+export type ButtonProps = {
+
+  /**
+   * The children nodes to be rendered with the button context.
+   */
+  children?: ReactNode,
+
+  /**
+   * Defines the button hierarchy. Either:
+   *
+   * - primary: button associated with the most significant (and therefore primary) action on the page;
+   * - neutral: button associated with a general purpose action;
+   * - danger: button associated with a potentially dangerous action.
+   */
+  hierarchy?: ButtonHierarchies,
+
+  /**
+   * Redirect url of a link button.
+   */
+  href?: string,
+
+  /**
+   * Sets an icon component for the button.
+   */
+  icon?: ReactNode,
```

Comment:
Qui non dovremmo mettere il tipo di Icon componente | ReactNode ?

Reply:
I get the error `Type 'IconType' is not assignable to type 'ReactNode'` 🥲

Reply:
We could not solve the problem and we left ReactNode. In the future, it would be nice to use our interface